httpx/utils.py

```diff
@@ -169,3 +169,12 @@ def get_logger(name: str) -> logging.Logger:
             logger.addHandler(handler)
 
     return logging.getLogger(name)
+
+
+def safe_encode(str_or_bytes: typing.Union[str, bytes]) -> bytes:
```

Comment:
Can we change this to `to_bytes()` to match `six`? Also this is almost certainly applicable in other places in the codebase, let's open a separate small PR to apply that change? :)